src/serac/physics/utilities/variational_form/polynomials.hpp

```diff
@@ -0,0 +1,298 @@
+};
+// clang-format on
+
+constexpr int factorial(int n)
+{
+  int nfactorial = 1;
+  for (int i = 2; i <= n; i++) {
+    nfactorial *= i;
+  }
+  return nfactorial;
+}
+
+template <int n, typename T>
+constexpr tensor<T, n> powers(T x)
+{
+  tensor<T, n> values{};
+  values[0] = T(1.0);
+  for (int i = 1; i < n; i++) {
+    values[i] = values[i - 1] * x;
+  }
+  return values;
+}
+
+/**
+ * @brief Chebyshev polynomials of the first kind
+ * Satisfying:
+ * T_n(cos(t)) == cos(n*t)
+ */
+template <int n, typename S>
+constexpr tensor<S, n> ChebyshevT(S x)
+{
+  tensor<S, n> T{};
+
+  if (0 < n) T[0] = 1.0;
+  if (1 < n) T[1] = x;
+  for (int i = 2; i < n; i++) {
+    T[i] = 2 * x * T[i - 1] - T[i - 2];
+  }
+
+  return T;
+}
+
+/**
+ * @brief Chebyshev polynomials of the second kind
+ * Satisfying:
+ * sin(t) U_n(cos(t)) == sin((n+1)*t)
+ */
+template <int n, typename T>
+tensor<T, n> ChebyshevU(T x)
+{
+  tensor<T, n> U{};
+
+  if (0 < n) U[0] = 1.0;
+  if (1 < n) U[1] = 2 * x;
+  for (int i = 2; i < n; i++) {
+    U[i] = 2 * x * U[i - 1] - U[i - 2];
+  }
+
+  return U;
+}
+
+/**
+ * Legendre Polynomials, orthogonal on the domain (-1, 1)
+ * with unit weight function.
+ */
+template <int n, typename T>
+tensor<T, n> Legendre(T x)
+{
+  tensor<T, n> P{};
+
+  if (0 < n) P[0] = 1.0;
+  if (1 < n) P[1] = x;
+  for (int i = 2; i < n; i++) {
+    P[i] = ((2 * i - 1) * x * P[i - 1] - (i - 1) * P[i - 2]) / T(i);
+  }
+
+  return P;
+}
+
+template <int n, typename T>
+tensor<T, n> Bernstein(T s)
+{
+  tensor<T, n> B;
+
+  T t = 1.0 - s;
+
+  T f = 1.0;
+  for (int i = 0; i < n; i++) {
+    B[i] = f;
+    f *= s / (i + 1);
+  }
+
+  f = factorial(n - 1);
+  for (int i = 0; i < n; i++) {
+    B[n - i - 1] *= f;
+    f *= t / (i + 1);
+  }
+
+  return B;
+}
+
+template <int n, typename T>
+tensor<T, n> GaussLobattoInterpolation(T x)
+{
+  if constexpr (n == 2) {
+    return {0.5 * (1.0 - x), 0.5 * (1.0 + x)};
+  }
+  if constexpr (n == 3) {
+    return {0.5 * x * (x - 1.0), 1.0 - x * x, 0.5 * x * (x + 1.0)};
+  }
+  if constexpr (n == 4) {
+    static constexpr double s = 2.23606797749978981;
+    return {-0.125 * (x - 1.0) * (5.0 * x * x - 1.0), 0.625 * (s * x - 1.0) * (x * x - 1.0),
+            -0.625 * (s * x + 1.0) * (x * x - 1.0), 0.125 * (x + 1.0) * (5.0 * x * x - 1.0)};
+  }
+
+  return tensor<T, n>{};
+}
+
+template <int n, typename T>
+tensor<T, n> GaussLobattoInterpolationDerivative(T x)
+{
+  if constexpr (n == 2) {
+    return {-0.5, 0.5};
+  }
+  if constexpr (n == 3) {
+    return {x - 0.5, -2.0 * x, x + 0.5};
+  }
+  if constexpr (n == 4) {
+    static constexpr double s = 2.23606797749978981;
+    return {0.125 * (1.0 + 5.0 * (2.0 - 3.0 * x) * x), 0.125 * (-5.0 * (s + x * (2.0 - 3.0 * s * x))),
+            0.125 * (5.0 * (s - x * (2.0 + 3.0 * s * x))), 0.125 * (-1.0 + 5.0 * x * (2 + 3.0 * x))};
+  }
+
+  return tensor<T, n>{};
+}
+
+template <int n, typename T>
+constexpr tensor<T, n> GaussLobattoInterpolation01(T x)
```

Comment:
Comments are needed explaining the difference between `GaussLobattoInterpolation01` and `GaussLobattoInterpolation`. I assume this is just reference intervals from [0,1] vs. [-1, 1].